From 2c52da67caf571f0bcacfdadfb692b6f2a4deff9 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Sun, 4 Aug 2024 20:46:50 +0530
Subject: [PATCH] Changes to compile libwifihal with latest Android

Tests done:
- Build with libwifi-hal-intel enabled
- Wi-Fi on/off/scan

Tracked-On: OAM-123043
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 Android.bp             | 78 ++++++++++++++++++++++++++++++++++++++++++
 Android.mk             | 57 ------------------------------
 include/wifi_hal_ext.h |  2 +-
 lib/driver_if.h        |  6 +---
 util/utils.h           |  3 +-
 5 files changed, 81 insertions(+), 65 deletions(-)
 create mode 100644 Android.bp

diff --git a/Android.bp b/Android.bp
new file mode 100644
index 0000000..448dce5
--- /dev/null
+++ b/Android.bp
@@ -0,0 +1,78 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+package {
+    // See: http://go/android-license-faq
+    default_applicable_licenses: [
+        "Android-Apache-2.0",
+    ],
+}
+
+soong_config_module_type_import {
+    from: "frameworks/opt/net/wifi/libwifi_hal/Android.bp",
+    module_types: ["wifi_cc_defaults"],
+}
+
+cc_library_static {
+    name: "libwifi-hal-intel",
+
+    cflags: [
+        "-Wno-int-to-pointer-cast",
+        "-D_FORTIFY_SOURCE=2",
+        "-fstack-protector-strong",
+        "-fno-delete-null-pointer-checks",
+        "-fwrapv",
+        "-Wall",
+        "-Werror",
+        "-Wno-format",
+        "-Wno-reorder",
+        "-Wno-unused-function",
+        "-Wno-unused-parameter",
+        "-Wno-unused-private-field",
+        "-Wno-unused-variable",
+        "-Wformat-security",
+        "-Wextra",
+        "-Wsign-compare",
+        "-Wpointer-arith",
+        "-Wcast-qual",
+        "-Wcast-align",
+        "-Wno-int-to-pointer-cast",
+        "-Wno-missing-field-initializers",
+        "-Wno-conversion-null",
+        "-Wnull-dereference",
+        "-DCONFIG_LIBNL20",
+        "-DCONFIG_ANDROID_LOG",
+    ],
+
+    local_include_dirs: [
+        "include",
+        "util",
+    ],
+
+    include_dirs: [
+        "external/libnl/include",
+    ],
+
+    header_libs: [
+        "libutils_headers",
+        "libhardware_headers",
+        "wifi_legacy_headers",
+        "device_kernel_headers",
+    ],
+
+    shared_libs: [
+        "liblog",
+        "libcutils",
+        "libdl",
+        "libnl",
+    ],
+
+    srcs: [
+        "lib/wifi_hal.cpp",
+        "util/hal_debug.cpp",
+        "lib/driver_if.cpp",
+    ],
+
+    proprietary: true,
+}
+
diff --git a/Android.mk b/Android.mk
index 0aef1bd..59247e9 100644
--- a/Android.mk
+++ b/Android.mk
@@ -2,61 +2,4 @@ LOCAL_PATH := $(call my-dir)
 
 include $(CLEAR_VARS)
 
-include $(LOCAL_PATH)/android.config
-
-INCLUDE_DIR := include
-UTIL_DIR := util
-
-LOCAL_REQUIRED_MODULES :=
-
-LOCAL_CPPFLAGS += -Wno-unused-parameter -Wno-int-to-pointer-cast -Wno-missing-field-initializers
-LOCAL_CPPFLAGS += \
-        -D_FORTIFY_SOURCE=2 \
-        -fstack-protector-strong \
-        -fno-delete-null-pointer-checks \
-        -fwrapv \
-        -Wformat -Wformat-security \
-        -Wall -Wextra -Wsign-compare -Wpointer-arith \
-        -Wcast-qual -Wcast-align \
-        -Wno-unused-parameter \
-        -Wno-int-to-pointer-cast \
-        -Wno-missing-field-initializers \
-        -Wno-conversion-null \
-        -Werror \
-        -Wnull-dereference
-LOCAL_CPPFLAGS += -DCONFIG_LIBNL20
-LOCAL_CPPFLAGS += -DCONFIG_ANDROID_LOG
-
-ifneq ($(wildcard external/libnl),)
-LOCAL_C_INCLUDES += external/libnl/include
-else
-LOCAL_C_INCLUDES += external/libnl-headers
-endif
-
-LOCAL_C_INCLUDES += \
-        $(call include-path-for, libhardware_legacy)/hardware_legacy \
-	$(LOCAL_PATH)/$(UTIL_DIR) \
-	$(LOCAL_PATH)/$(INCLUDE_DIR) \
-
-LOCAL_SRC_FILES := \
-        lib/wifi_hal.cpp \
-	$(UTIL_DIR)/hal_debug.cpp \
-	lib/driver_if.cpp \
-
-LOCAL_SHARED_LIBRARIES := liblog libcutils libdl
-LOCAL_HEADER_LIBRARIES += libutils_headers libhardware_headers
-
-ifdef CONFIG_DEBUG
-ifeq ($(shell expr $(CONFIG_DEBUG) : '[0-3]$\'), 1)
-LOCAL_CPPFLAGS += -DCONFIG_DEBUG=$(CONFIG_DEBUG)
-else
-$(error CONFIG_DEBUG must be an integer between 0 and 3)
-endif
-endif
-
-LOCAL_MODULE := libwifi-hal-intel
-LOCAL_PROPRIETARY_MODULE := true
-
-include $(BUILD_STATIC_LIBRARY)
-
 include $(LOCAL_PATH)/wpa_supplicant_8_lib/Android.mk
diff --git a/include/wifi_hal_ext.h b/include/wifi_hal_ext.h
index f2e293a..1e4585e 100644
--- a/include/wifi_hal_ext.h
+++ b/include/wifi_hal_ext.h
@@ -41,8 +41,8 @@
 
 #include <net/if.h>
 #include <pthread.h>
+#include <hardware_legacy/wifi_hal.h>
 #include "utils.h"
-#include "wifi_hal.h"
 #include "ieee802_11_defs.h"
 
 struct wifi_interface_info {
diff --git a/lib/driver_if.h b/lib/driver_if.h
index 0abdef9..acd0455 100644
--- a/lib/driver_if.h
+++ b/lib/driver_if.h
@@ -17,11 +17,7 @@
 #ifndef __DRIVER_IF_H__
 #define __DRIVER_IF_H__
 
-#ifdef ANDROID
-#include <wifi_hal.h>
-#else
-#include "wifi_hal.h"
-#endif
+#include <hardware_legacy/wifi_hal.h>
 #include "ieee802_11_defs.h"
 #include "utils.h"
 
diff --git a/util/utils.h b/util/utils.h
index 39e603a..6777570 100644
--- a/util/utils.h
+++ b/util/utils.h
@@ -19,9 +19,8 @@
 
 #include <stdlib.h>
 #include <string.h>
-
+#include <hardware_legacy/wifi_hal.h>
 #include "list.h"
-#include "wifi_hal.h"
 
 #ifndef BIT
 #define BIT(_x) (1 << (_x))
-- 
2.45.2

